### DISCUSSION TITLE
Ability to catch all errors like errorCaptured

### DISCUSSION THREAD

**Discussion by ghost** [vuejs/pinia#2471] [Help and Questions] [answered]
I'm trying to set up global error handling in my app, which I can partially accomplish with [errorCaptured](https://vuejs.org/api/options-lifecycle.html#errorcaptured).  Unfortunately, this does not catch any errors that are thrown from Pinia... or rather, I might not know how to tie into it.  Is there something like this that exists in Pinia?  I know I can add an interceptor in axios or whatever request library, but it's possible that the error occurs before or after the request.

Basically in [this example](https://codesandbox.io/p/sandbox/funny-mcnulty-dzkq75?file=%2Fsrc%2FApp.vue%3A8%2C22), I'd like to show the Pinia errors in the Errors heading below the components.  The error is being thrown from the stores/test file, which does not get caught by errorCaptured.

**Comment by ghost** [comment, marked as answer]
I may have figured it out... I need to tap into both [error](https://developer.mozilla.org/en-US/docs/Web/API/Window/error_event) and [unhandledrejection](https://developer.mozilla.org/en-US/docs/Web/API/Window/unhandledrejection_event).  The latter is to handle promises and the former is for everything else... I would've needed the latter because errorCaptured does not actually capture promise errors, which is the majority of what I need to catch.
